### PR TITLE
HTTP Tracing

### LINKED ISSUE #26257: HTTP tracing development

### PR DESCRIPTION
**Pull request by fmhwong**
Fixes #26257 

HTTP tracing is currently under beta fencing.

### CODE REVIEW

**Comment by fmhwong** [comment]
#build

**Comment by fmhwong** [comment]
#build

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/SpanCurrentServlet.java [hunk 1]
```diff
@@ -24,10 +24,11 @@
...
-@WebServlet("/SpanCurrentServlet")
+@WebServlet("/testSpanCurrent")
 public class SpanCurrentServlet extends FATServlet {
 
-    @Test
+    // With HTTP tracing enabled, there will be always a current span.  Comment out this test
+    // @Test
     public void testGetCurrentSpan_Default() {
         Span span = Span.current();
         assertEquals(span, Span.getInvalid()); //Current span has no context as none was created
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/SpanCurrentServlet.java]: Could we replace this with a test that runs a task on an executor without propagating the context and checks the current span there?

File: dev/io.openliberty.io.opentelemetry/bnd.bnd [hunk 2]
```diff
@@ -41,7 +41,7 @@ Import-Package: \
 
 Export-Package: \
   io.opentelemetry.extension.trace.propagation;version=${openTelemetryVersion};thread-context=true,\
-  io.opentelemetry.*;version=${openTelemetryVersion}
+  io.opentelemetry.*;version=${openTelemetryVersion};thread-context=true
```
Comment (Azquelt) [inline dev/io.openliberty.io.opentelemetry/bnd.bnd:44]: Why is this needed? I don't think it should be necessary.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10Servlet/server.xml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    </featureManager>
+
+    <application id="TelemetryServletTestApp" name="TelemetryServletTestApp"
+        type="war" location="TelemetryServletTestApp.war">
+        <classloader apiTypeVisibility="+third-party" />
+    </application>
+    
+    <logging traceSpecification="TELEMETRY=all"/>
+
+    <javaPermission className="java.net.URLPermission" name="http://localhost:-/-" actions="GET:" />
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10Servlet/server.xml:20]: Please can we add a comment to say why this permission is needed?

I assume it's because the test is using HttpUrlConnection, but it'll be helpful to know why it's there if we are looking at the server.xml in future and aren't sure if it's still needed.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/SpanCurrentServlet.java (hunk 1 above)
Comment (fmhwong) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/SpanCurrentServlet.java]: There is an internal property to turn off the `TelemetryServletFilter`.  Do you want to keep all tests in this server as is and I can add the property to this server.

File: dev/io.openliberty.io.opentelemetry/bnd.bnd (hunk 2 above)
Comment (fmhwong) [reply dev/io.openliberty.io.opentelemetry/bnd.bnd:44]: This is needed or I would get this error
```
[ERROR   ] CWMOT5002E: An internal error occurred in MicroProfile Telemetry. The error is: io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException: OTLP gRPC Trace Exporter enabled but opentelemetry-exporter-otlp not found on classpath. Make sure to add it as a dependency to enable this feature.class not found: io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter.
```

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10Servlet/server.xml (hunk 3 above)
Comment (fmhwong) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10Servlet/server.xml:20]: Yes, that is for the `HttpUrlConnection` and I will add a comment.

**Comment by fmhwong** [comment]
#build

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/SpanCurrentServlet.java (hunk 1 above)
Comment (fmhwong) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/SpanCurrentServlet.java]: Revised it to under in a task on an executor.

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10Servlet/server.xml (hunk 3 above)
Comment (fmhwong) [reply dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/publish/servers/Telemetry10Servlet/server.xml:20]: Comment added

File: dev/io.openliberty.io.opentelemetry/bnd.bnd (hunk 2 above)
Comment (Azquelt) [reply dev/io.openliberty.io.opentelemetry/bnd.bnd:44]: This is concerning. If that were the case then we should have far more 1.0 tests failing since our existing collector tests rely on the OTLP exporter.

Comment (fmhwong) [reply dev/io.openliberty.io.opentelemetry/bnd.bnd:44]: I have created another PR https://github.com/OpenLiberty/open-liberty/pull/26278.  We can continue to discuss there.

**Comment by fmhwong** [comment]
This is superseded by https://github.com/OpenLiberty/open-liberty/pull/26278
